### PR TITLE
helm: Add smoke-test job for `helm test`

### LINKED ISSUE #2060: helm: run a write / read smoke test
#### Is your feature request related to a problem? Please describe.

After installing Mimir via the helm chart, I'd like to know whether time series ingestion and querying is functional. There are a few ways to validate it, but it would be nice if it was integrated into `helm test`.

#### Describe the solution you'd like

Use the new smoke test mode from #2047 to run a basic test against helm-installed Mimir clusters.

### LINKED ISSUE #2109: [mimir-continuous-test]: Improve smoke-test logging
#### Is your feature request related to a problem? Please describe.

Currently a successful smoke test logs the following:

```
2022-06-14T16:50:50.5568572Z level=info ts=2022-06-14T16:50:44.661318442Z caller=write_read_series.go:79 test=write-read-series msg="Finding previously written samples time range to recover writes and reads from previous run"
2022-06-14T16:50:50.5570373Z level=info ts=2022-06-14T16:50:44.685304729Z caller=write_read_series.go:83 test=write-read-series msg="No valid previously written samples time range found"
```

This doesn't look like a successful test with language like "no valid ... found".

#### Describe the solution you'd like

When smoke-test mode is enabled, the test should log more informative messages on success. For example, what tests were run, how many samples were written and read successfully, etc.

#### Describe alternatives you've considered

We could instead document this as an expected output, which would be less discoverable than improving the message.

#### Additional context

These smoke tests are used in the `helm test` command, so it's likely the first interaction some users will have with writing / reading samples from Mimir after initially installing it.

### PR DESCRIPTION
**Pull request by Logiraptor**
#### What this PR does

This PR adds a Job definition to the helm chart with the annotation `helm.sh/hook: test`. This Job will be applied when running `helm test` and also when running the `ct` tool in our CI. 

The intention is to run a basic write / read smoke test in the helm chart CI. 

#### Which issue(s) this PR fixes or relates to

Fixes #2060
Fixes #2109 

#### Checklist

- [ ] Update smoke-test image reference to weekly r190 once it's available
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by Logiraptor** [comment]
Will update this PR once weekly r190 is available: it contains the necessary feature in `mimir-continuous-test`. Should be good to go after that.

File: operations/helm/charts/mimir-distributed/values.yaml [hunk 1]
```diff
@@ -24,6 +24,20 @@ image:
   # pullSecrets:
   #   - myRegistryKeySecretName
 
+# Settings for the smoke-test job.
```
[the remark this answers is not in the document]
Comment (Logiraptor) [reply operations/helm/charts/mimir-distributed/values.yaml:27]: We could add it to the getting started doc, but I don't think it's likely users will need to modify the settings. 

I think we should probably add it to the getting started once we have it working with enterprise auth enabled. I'll move this section to the bottom.

**Comment by Logiraptor** [comment]
@krajorama Yeah that is the output for a successful run. 100% agree it's not user-friendly. It's a leftover from the fact that `mimir-continuous-test` is designed to run forever so it doesn't log much unless something goes wrong. I just created an issue for us to improve the logging: https://github.com/grafana/mimir/issues/2109

**Comment by Logiraptor** [comment]
@krajorama I made an update to the continuous-test code. The output for a successful test is now:

```
level=info ts=2022-06-15T16:50:56.803065628Z caller=write_read_series.go:79 test=write-read-series msg="Finding previously written samples time range to recover writes and reads from previous run"
level=info ts=2022-06-15T16:50:56.832870129Z caller=write_read_series.go:83 test=write-read-series msg="No valid previously written samples time range found, will continue writing from the nearest interval-aligned timestamp"
level=info ts=2022-06-15T16:50:56.976657781Z caller=manager.go:75 msg="Test passed" test=write-read-series
```

**Review by krajorama — APPROVED** [review]
LGTM

**Pull request merged**